lib/realtime/rate_counter/rate_counter.ex

```diff
@@ -138,35 +141,42 @@ defmodule Realtime.RateCounter do
   def handle_info(:tick, state) do
     Process.cancel_timer(state.tick_ref)
 
-    {:ok, count} = GenCounter.get(state.id)
-    :ok = GenCounter.put(state.id, 0)
+    state =
+      case GenCounter.get(state.id) do
```

Comment:
We can have the RateCounter monitor the GenCounter and gracefully terminate and we won't have to even have a case statement here.